tests/src/Functional/GroupUpdateTest.php

```diff
@@ -0,0 +1,190 @@
+    $this->createContentType(['type' => 'content_group']);
+    Og::groupManager()->addGroup('node', 'content_group');
+
+    // Create a role with only the 'update group' permission.
+    $content_editor_role = OgRole::create();
+    $content_editor_role
+      ->setName('content_editor')
+      ->setLabel('Content group editor')
+      ->setGroupType('node')
+      ->setGroupBundle('content_group')
+      ->grantPermission(OgAccess::UPDATE_GROUP_PERMISSION)
+      ->save();
+
+    // Create a group content owned by the group owner.
+    $values = [
+      'title' => 'My awesome content group',
+      'type' => 'content_group',
+      'uid' => $this->group_owner->id(),
+      'status' => 1,
+    ];
+    $this->content_group = Node::create($values);
+    $this->content_group->save();
+
+    // Subscribe the editor user to the groups.
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->group_editor->id())
+      ->setEntityId($this->content_group->id())
+      ->setGroupEntityType($this->content_group->getEntityTypeId())
+      ->setRoles([$content_editor_role->id()])
+      ->save();
+  }
+
+  /**
+   * Setup dummy entity_test group entity and appropriate og permissions.
+   */
+  public function setUpEntityTestEntity() {
+    // Create an entity_test bundle called 'entity_group' and make it
+    // an Og group.
+    entity_test_create_bundle('entity_group');
+    Og::groupManager()->addGroup('entity_test', 'entity_group');
+
+    // Create a role with only the 'update group' permission.
+    $entity_editor_role = OgRole::create();
+    $entity_editor_role
+      ->setName('entity_editor')
+      ->setLabel('Entity group editor')
+      ->setGroupType('entity_test')
+      ->setGroupBundle('entity_group')
+      ->grantPermission(OgAccess::UPDATE_GROUP_PERMISSION)
+      ->save();
+
+    // Create an entity_test entity owned by the group owner.
+    $values = [
+      'title' => 'My awesome group',
+      'type' => 'entity_group',
+      'uid' => $this->group_owner->id(),
+    ];
+    $this->entity_group = EntityTest::create($values);
+    $this->entity_group->save();
+
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->group_editor->id())
+      ->setEntityId($this->entity_group->id())
+      ->setGroupEntityType($this->entity_group->getEntityTypeId())
+      ->setRoles([$entity_editor_role->id()])
+      ->save();
+  }
+
+  /**
+   * Tests 'update group' special permission.
+   *
+   * @dataProvider ogUpdateAccessProvider
+   */
+  public function testUpdateAccess($entity) {
+    // The editor should have permissions due to the 'update group' special
+    // permission.
+    $this->drupalLogin($this->group_editor);
+    $this->drupalGet($this->{$entity}->toUrl('edit-form'));
+    $this->assertSession()->statusCodeEquals(200);
```

Comment:
Well, I'm waiting Amitai's review because this is my first impression. There is one thing blank for me though: what is the permission needed to delete the group and is it part of og?
Maybe 'update group' refers to both update and delete as the only other permission related to groups is administer group.
There are already tests to check the consistency of the permissions (like retrieving and checking if the user indeed has these permissions) but in a real case scenario, it wasn't working due to missing mapping of the permission. That's why I thought a browser test would be a good way of testing it.